modules/javafx.graphics/src/main/java/com/sun/glass/ui/Window.java

```diff
@@ -1322,7 +1322,7 @@ protected void notifyFocus(int event) {
 
         if (this.isFocused != focused) {
             this.isFocused = focused;
-            if (this.isFocused) {
+            if (this.isFocused && this.isVisible) {
```

Comment:
On my Window10 machine, with this change, `Window.focusedWindow` remains `null` even after the first window is shown onto the screen and is focused. And It continues to remain `null` until some mouse or key action is performed on the window.
I am not sure if this causes any side effects. It looks like the `Window.focusedWindow` is mostly(only) used for Monocle.
Can you please confirm the behavior that `Window.focusedWindow` remain `null` and check for any side effects.

Reply:
As mentioned - I don't have a good setup to test this code on Windows.

But I've checked where focusedWindow/getFocusedWindow is used, and I can verify your assumption. I've searched through the whole project and the variable is only used in the MonocleCode.

The fact that focusedWindow get's sometimes set is probably the cause of the irregular happening memoryleak on Window.

Reply:
The motivation for this change was, that I've seen a similar bug on native-windows without monocle. 
In this case a leak happened related to the focusedWindow variable. 
Sadly I don't have a test for this bug, because it is undeterministic. 
For that reason, I thought it would make sense to keep this change, because I think it fixes it.
If I remember correctly, it was related to Stages which were focused but closed at the same time, which only weren't collected because of the focusedWindow variable.
Because the variable also isn't used on NativeWindows it doesn't seem to do much harm.

I've now removed it, but it might be considered to be readded.